src/components/avatar/Avatar.types.tsx

```diff
@@ -0,0 +1,18 @@
+import { ViewProps } from 'react-native'
+
+import { sizes } from './Avatar'
+import { IllustrationName } from './Illustrations.types'
+
+export interface AvatarProps extends Omit<ViewProps, 'children'> {
+  src: IllustrationName
+  /**
+   * @default 'md'
```

Comment:
q: What's the advantage of this `@default` VS `Avatar.defaultProps = { size: 'md' }`?

Reply:
@default is purely about documentation, it doesn't do anything. It's not super necessary as well tbh, it can be nice to know if a prop has a default value and if it does which one it is

<img width="659" alt="Screenshot 2023-06-17 at 16 47 33" src="https://github.com/cratera/papers-game/assets/63152281/82fd9e40-34f4-40f6-b703-78c98149ace9">

Reply:
Ah, so I think there's a bug in the codebase, give me a few minutes to try it out.

Reply:
Okay, this one is tricky to explain. Let me try:

1. When you say "is purely about documentation", this seems fragile, because in `Avatar.ts`, you declare a default value on line 21: `size = 'md'`. If you change this line to another value, then the docs gets outdated (without any warning).
2. You say this prop is optional `size?: keyof typeof sizes`, but it is not, it's required, otherwise the component would break. The workaround for DX is to pass the default value in the props. If you remove the `?`, but keep the default value, [TS will complain](https://github.com/cratera/papers-game/assets/14869087/6f28a9e8-c021-4bd5-b747-a099f36b420f).

How to solve this? The solution I've learned is using [defaultProps](https://github.com/cratera/papers-game/assets/14869087/17844f7a-f8b3-4e30-affe-adf14bcefe9d). 

```js
Avatar.defaultProps = {
  size: 'md',
} as AvatarProps
```

I've learned this recently from this [react/require-default-props](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-default-props.md).


I don't know if it's the best way, but it's the one I know (reading more about it right now)

Reply:
Again, this might be overengineering, as I agree that destruct `size = 'md'` is way easier. But knowing that the `@default 'md'` is so far away (in a different file), and can get stale without warning, made me question it. I realized the trade-off is the lack of docs in the auto-complete.  So, another scenario where I'm not sure about the best answer.

Reply:
I agree that the whole `@default` thing is fragile and might not add much value honestly, but I'm confused as to why you think it's broken as it is atm. The prop is meant to be optional because I set a default in the component itself, i feel like the way you suggest to go about it is just more verbose and it it's actually counterintuitive to have a required prop that you actually don't have to pass, wouldn't you agree?

Reply:
It's not broken, sorry, it's just the concept of optional/required that I've mixed.

> it's counterintuitive to have a required prop that you actually don't have to pass

Agreed 🙃 That's the whole definition of optional. An optional always has a default value (even if it's null/false). A required doesn't has a default value. I shall get back to my other projects to review this one. 💡 

Nevermind what I said! 🙇

Reply:
Just one more thing that I would like to add: you still have type safety in the sense that, when you're trying to use the optional prop and you didn't set a default value you will still get an error if it shouldn't be `undefined`

<img width="791" alt="Screenshot 2023-06-18 at 12 36 36" src="https://github.com/cratera/papers-game/assets/63152281/8d1fc41b-ee68-48c6-b565-a2f993c6045d">

Reply:
Oh, I see. In this case is because we are passing it to an object/array, so who's complaining is the Array construct, not the AvatarProps. In other simpler scenarios, it would "break" without errors.
<img width="390" alt="image" src="https://github.com/cratera/papers-game/assets/14869087/ddb8595e-4bde-41d7-b0dc-34f31f820ec0">

Reply:
Anyway, it was a good chat here, even though I just reviewed 1% of the code ahahahah

Reply:
Fair point, but in that example, there shouldn't be an error in the first place because any of those values are allowed inside the `p`, but yeah there might be some situations where this could become a problem potentially 🤔